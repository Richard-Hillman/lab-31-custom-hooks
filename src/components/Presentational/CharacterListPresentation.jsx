import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
// import CharacterDetailsPresentation from './CharacterDetailsPresentation';

function CharacterListPresentation({ characters }) {
  const characterElements = characters.map(character => (
    <li key={character.name}>
      <Link to={`/characters/${character.name}` }>
        <img src={character.image} />
        <h2>{character.name}</h2>
      </Link>
    </li>
  ));

  return (
    <ul data-testid="charactersTest">
      {characterElements}
    </ul>
  );
}

CharacterListPresentation.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    rank: PropTypes.string.isRequired
  })).isRequired
};

export default CharacterListPresentation;
