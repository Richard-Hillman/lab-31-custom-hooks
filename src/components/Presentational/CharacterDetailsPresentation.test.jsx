/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterListPresentation from './CharacterListPresentation';

describe('CharacterDetailsPresentation component', () => {
  afterEach(() => cleanup());
  it('renders CharacterListPresentation', () => {
    const { asFragment } = render(<CharacterListPresentation characters={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
