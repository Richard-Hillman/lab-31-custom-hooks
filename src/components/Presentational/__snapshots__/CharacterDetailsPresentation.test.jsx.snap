// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetailsPresentation component renders CharacterListPresentation 1`] = `
<DocumentFragment>
  <ul
    data-testid="charactersTest"
  />
</DocumentFragment>
`;
